File: Problem 2.py
import time


start = time.time() # Time start

a = 1
b = 2
c = 0
sum = 2

while c < 4000000: # Limit

    # Code to generate fibonacci sequence
    
    c = a + b

    if c % 2 == 0:
        sum = sum + c

    a = b
    b = c

end = time.time() # Time stop
length = end - start

print(sum)
print(f'It took {length} seconds.')

# Very simple program again.
# Output: 4613732

File: Problem 2.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(int limit){
	int a=1,b=2,sum=0;
	while(b<limit){
		if(b%2==0){
			sum += b;
		}
		int c = a+b;
		a = b;
		b = c;
	}
	return sum;
}

int main(){
	cout << solution(4000000) << endl;
	return 0;
}
